Add 10.0.2.2 to INTERNAL_IPS to support Django Debug Toolbar in Docker

willbarton opened the pull request:
This PR adds `10.0.2.2` to `INTERNAL_IPS` when `ENABLE_DEBUG_TOOLBAR` is set. This will allow usage of the Django Debug Toolbar in our Docker setup.

~~This PR makes the `INTERNAL_IPS` setting configurable to allow us to use Django Debug Toolbar in Docker when `ENABLE_DEBUG_TOOLBAR` is enabled.~~

~~This is a WIP because: @wpears @rosskarchner I'm not sure where the `10.0.2.2` IP address comes from, or whether it's specific to my machine. I discovered it by printing `request.META['REMOTE_ADDR']` in a request cycle to discover what IP requests from my Mac were coming from inside the python container. Is there a way we can make that information more easily discoverable? Is `10.0.2.2` consistent for everyone using our Docker setup?~~

## Testing

1. Add the following variable to your `.env` file:

   ```
   export ENABLE_DEBUG_TOOLBAR=''
   ```

2. Restart the `python` container
3. Visit http://localhost:8000 and observe that the Django Debug Toolbar is enabled.

## Todos

~~The source of the IP that needs to be added to enable the toolbar on docker needs to be documented.~~

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

acrewdson commented on cfgov/cfgov/settings/local.py:
```diff
@@ -40,7 +40,9 @@
 
-    INTERNAL_IPS = ('127.0.0.1',)
+    INTERNAL_IPS = [
+        i for i in os.environ.get('INTERNAL_IPS', '127.0.0.1').split(' ')
```
could this be written without the list comp and without the `split` arg, just: `os.environ.get('INTERNAL_IPS', '127.0.0.1').split()`?

chosak replied:
Any reason why we shouldn't just add `10.0.0.2` to this list (maybe with a comment informing what it is there for) and leave it at that? It wouldn't be needed in non-Docker environments but would be much simpler and would avoid needing to define this in `.env`.

rosskarchner commented:
Instead of messing with IP's, how about setting SHOW_TOOLBAR_CALLBACK to a function that returns True if DEBUG is true (ignoring INTERNAL_IPS)  https://django-debug-toolbar.readthedocs.io/en/latest/configuration.html

willbarton commented:
> Instead of messing with IP's, how about setting SHOW_TOOLBAR_CALLBACK to a function that returns True if DEBUG is true (ignoring INTERNAL_IPS) https://django-debug-toolbar.readthedocs.io/en/latest/configuration.html

@rosskarchner I like the simplicity of this solution, but I'm not sure I want the toolbar on all the time. It *does* slow down local requests quite a lot.

willbarton commented:
@rosskarchner Oh! Right! Yes, it would! 

I'll do exactly what you suggest.

willbarton commented:
I've decided to go as simple as possible and add `10.0.2.2` to `INTERNAL_IPS`. If this becomes problematic, we can explore other alternatives. I've updated the PR accordingly.

I also added `ENABLE_DEBUG_TOOLBAR` to `.env_SAMPLE` and left it commented out.
